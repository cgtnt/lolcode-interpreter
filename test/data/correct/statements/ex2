HAI

I HAS A fish ITZ "bob"
I HAS A stone
I HAS A fishtwo ITZ A YARN

VISIBLE fish!
stone R "two"
VISIBLE stone!
fishtwo R "lol"
VISIBLE fishtwo!

KTHXBYE
