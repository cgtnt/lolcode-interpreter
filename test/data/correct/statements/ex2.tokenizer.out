@1:T_BEGIN:HAI-@3:T_COMMAND_TERMINATOR:
-@3:T_DECLARE_VAR:I HAS A-@3:T_T_IDENTIFIER:fish-@3:T_DECLARE_SET_VAR:ITZ-@3:T_T_STRING:"bob"-@4:T_COMMAND_TERMINATOR:
-@4:T_DECLARE_VAR:I HAS A-@4:T_T_IDENTIFIER:stone-@5:T_COMMAND_TERMINATOR:
-@5:T_DECLARE_VAR:I HAS A-@5:T_T_IDENTIFIER:fishtwo-@5:T_DECLARE_TYPE_VAR:ITZ A-@5:T_TI_STRING:YARN-@7:T_COMMAND_TERMINATOR:
-@7:T_WRITE_STDOUT:VISIBLE-@7:T_T_IDENTIFIER:fish-@7:T_BANG:!-@8:T_COMMAND_TERMINATOR:
-@8:T_T_IDENTIFIER:stone-@8:T_ASSIGN:R-@8:T_T_STRING:"two"-@9:T_COMMAND_TERMINATOR:
-@9:T_WRITE_STDOUT:VISIBLE-@9:T_T_IDENTIFIER:stone-@9:T_BANG:!-@10:T_COMMAND_TERMINATOR:
-@10:T_T_IDENTIFIER:fishtwo-@10:T_ASSIGN:R-@10:T_T_STRING:"lol"-@11:T_COMMAND_TERMINATOR:
-@11:T_WRITE_STDOUT:VISIBLE-@11:T_T_IDENTIFIER:fishtwo-@11:T_BANG:!-@13:T_COMMAND_TERMINATOR:
-@13:T_END:KTHXBYE-@14:T_COMMAND_TERMINATOR:
-@14:T_EOF:
